src/widget/wlibrarysidebar.cpp

```diff
@@ -205,6 +205,28 @@ void WLibrarySidebar::keyPressEvent(QKeyEvent* event) {
     //    // encoder click via "GoToItem"
     //    qDebug() << "GoToItem";
     //    TODO(xxx) decide what todo here instead of in librarycontrol
+    } else if (event->key() == Qt::Key_Left) {
+        auto selModel = selectionModel();
+        QModelIndexList selectedRows = selModel->selectedRows();
+        if (selectedRows.isEmpty()) {
+            return;
+        }
+        // If an expanded item is selected let QTreeView collapse it
+        QModelIndex selIndex = selectedRows.first();
+        if (!selIndex.isValid()) {
```

Comment:
The first index should always be valid. Why not? If unsure wrap it in VERIFY_OR_DEBUG_ASSERT.

Reply:
Sure, it _should_ I don't know what could happen behind the scenes. in wlibrarysidebar the index is verified for like every interaction.
I'll use the debug assert if you prefer that.

Reply:
It's a development version. Be as restrictive as possible until issues are reported.

Reply:
Let's simple add a DEBUG_ASSERT and continue without an early return. Both isExpanded() and parent() should not crash Mixxx and everything will work as expected.

Reply:
alight. done.